### PR TITLE
SAK-41898 - Error saving T&Q responses for other questions after nume…

### PR DESCRIPTION
**Pull request by mpellicer**
…ric validation failure

### CODE REVIEW

**Comment by mpellicer** [comment]
Please don't merge this yet, I need to fix an existing problem in the backend validation.

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java [hunk 1]
```diff
@@ -989,6 +989,7 @@ public int compare(ItemGradingData o1, ItemGradingData o2) {
                                totalItems, fibEmiAnswersMap, emiScoresMap, publishedAnswerHash, regrade, calcQuestionAnswerSequence);
         }
         catch (FinFormatException e) {
+			log.warn("Fin Format Exception while processing response. ", e);
```
Comment (adrianfish) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java]: Indentation a bit wrong here.

File: samigo/samigo-app/src/webapp/jsf/delivery/item/deliverFillInNumeric.jsp [hunk 2]
```diff
@@ -153,5 +152,50 @@ window.onload = function() {
...
 	});
 };
+
+includeWebjarLibrary('mathjs');
+var finFormatError = '<h:outputText value="#{deliveryMessages.fin_invalid_characters_error}" escape="false"/>';
+
+$( document ).ready(function() {
+	$('.fillInNumericInput').each(function() {
+		$(this).change(function(){
+			if(!$(this).val()){
```
Comment (adrianfish) [inline samigo/samigo-app/src/webapp/jsf/delivery/item/deliverFillInNumeric.jsp]: Don't need to wrap 'this' in a jquery object, just use this.value to get the input value.

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java (hunk 1 above)
Comment (mpellicer) [reply samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java]: Thanks!

File: samigo/samigo-app/src/webapp/jsf/delivery/item/deliverFillInNumeric.jsp (hunk 2 above)
Comment (mpellicer) [reply samigo/samigo-app/src/webapp/jsf/delivery/item/deliverFillInNumeric.jsp]: Please take a look at my most recent commit, thanks again!

**Pull request merged**